src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,43 +1,62 @@
 import React from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
-
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
-
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
-          </header>
-
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
-
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+type Props = {
+  user: User | null;
+  todo: Todo | null;
+  loader: boolean;
+  getClose: () => void;
+};
 
-              {' by '}
+export const TodoModal: React.FC<Props> = ({
+  user,
+  todo,
+  loader,
+  getClose,
+}) => (
+  <div className="modal is-active" data-cy="modal">
+    <div className="modal-background" />
 
-              <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-            </p>
+    {loader || !user || !todo ? (
```

Comment:
The condition `loader || !user || !todo` is used to decide whether to show the loader. This is correct, but ensure that the logic aligns with the expected behavior: the loader should display when data is being fetched or if the data is incomplete.